Dockerfiles/Cpp/Dockerfile

```diff
@@ -0,0 +1,105 @@
+ADD --chown=4000 https://raw.githubusercontent.com/eclipse-velocitas/vehicle-app-cpp-sdk/main/.scripts/common.sh                    /container/cpp/.scripts/
+ADD --chown=4000 https://raw.githubusercontent.com/eclipse-velocitas/vehicle-app-cpp-sdk/main/install_dependencies.sh               /container/cpp/
+ADD --chown=4000 https://raw.githubusercontent.com/eclipse-velocitas/vehicle-app-cpp-sdk/main/requirements.txt                      /container/cpp/   
+
+COPY ./common/ /scripts/
+
+# Install conan
+WORKDIR /scripts
+RUN ./install-python.sh && pip3 install -r /container/cpp/requirements.txt
+
+# Build dependencies for target architecture 
+WORKDIR /container/cpp
+USER 4000
+WORKDIR /container/cpp/build
+WORKDIR /container/cpp
+ARG TARGETARCH
+RUN bash ./install_dependencies.sh -x $TARGETARCH
+
+# Remove build files for final image
+RUN conan remove --builds --src --force '*'
+
+
+##############################################################################
+# Final step (run on TARGET, QEMU if needed)
+##############################################################################
+
+FROM --platform=$TARGETPLATFORM ubuntu:22.04 as final
+
+RUN apt-get --fix-missing update && apt-get upgrade -y && \
+        apt-get install -y apt-utils openssh-client gnupg2 dirmngr iproute2 \
+        procps lsof htop net-tools psmisc manpages manpages-dev \
+        curl tree wget ca-certificates unzip init-system-helpers \
+        bzip2 zip nano vim-tiny less jq lsb-release git zsh pigz \
+        apt-transport-https dialog libgcc1 libkrb5-3 iptables \
+        libgssapi-krb5-2 libicu[0-9][0-9] liblttng-ust[0-9] \
+        libstdc++6 zlib1g locales sudo ncdu man-db strace gnupg2 \
+        # install cpp packages
+        build-essential cmake clang lldb llvm gdb ninja-build cppcheck clang-format-14 clang-tidy-14 && \
+        # clean-up
+        apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* && \
+        # redirect format and tidy to the versioned variants
+        update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-14 100 && \
+        update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-14 100
+
+ADD https://raw.githubusercontent.com/devcontainers/features/main/src/common-utils/install.sh /container/scripts/install-common.sh
+RUN UID="4000" bash /container/scripts/install-common.sh 
+ADD https://raw.githubusercontent.com/devcontainers/features/main/src/docker-in-docker/install.sh /container/scripts/install-dind.sh
+RUN VERSION="latest" bash /container/scripts/install-dind.sh
+
+COPY ./common/ /scripts/
+
+WORKDIR /scripts
+RUN ./install-python.sh && \
+    ./install-dapr-cli.sh && \
+    ./install-k3d-prerequisites.sh \
+    ./install-k3d-tooling.sh
+
+ADD --chown=4000 https://raw.githubusercontent.com/eclipse-velocitas/vehicle-app-cpp-sdk/main/requirements.txt  /tmp/
+USER 4000
```

Comment:
We need to set the USER back to root at the end, otherwise we get errors when using this in our repo.

Reply:
Fixed.